hif/pcie/pcie.c

```diff
@@ -1320,8 +1320,8 @@ static void pcie_bf_mimo_ctrl_decode(struct mwl_priv *priv,
 			       &fp_data->f_pos);
 		filp_close(fp_data, current->files);
 	} else {
-		wiphy_err(priv->hw->wiphy, "Error opening %s! %x\n",
-			  filename, (unsigned int)fp_data);
+		wiphy_err(priv->hw->wiphy, "Error opening %s! %lx\n",
+			  filename, (long unsigned int)fp_data);
```

Comment:
`fp_data` here is an `ERR_PTR` (contains a negative err code).
You should use `PTR_ERR` to convert it to a `long` and print it with `%ld`.